dvc/scm/git.py

```diff
@@ -349,8 +345,7 @@ def install(self, use_pre_commit_tool=False):
             return
 
         config["repos"].append(entry)
-        with open(config_path, "w+") as fobj:
-            yaml.dump(config, fobj)
+        dump_yaml(config_path, config)
```

Comment:
Thinking of introducing a different API here to reduce boilerplate of create + load + dump (and, it's error-prone to work with parsers directly).
```python
with modify_yaml(config_path, tree=repo.tree) as config:
    config["repos"] = config.get("repos", [])
    config["repos"].append(entry)
```

Should also make `experiments` and `dvcfile` dumps nicer.